src/libs/actions/Report.js

```diff
@@ -869,7 +971,12 @@ function handleReportChanged(report) {
     // A report can be missing a name if a comment is received via pusher event
     // and the report does not yet exist in Onyx (eg. a new DM created with the logged in person)
     if (report.reportID && report.reportName === undefined) {
-        fetchChatReportsByIDs([report.reportID]);
+        updateIOUAndChatReportData(report.reportID);
+    }
+
+    // If this is a brand new IOU report, find and link corresponding chat report ID
+    if (report.reportName === CONST.REPORT.ACTIONS.TYPE.IOU && report.chatReportID === undefined) {
```

Comment:
Hmm so there are two ways to tell if a report is an IOU? Both the title and the `rNVP` - can we maybe pick one? This feels kind of like implementation details that we shouldn't have to worry about here.

Reply:
We don't have rNVP in this report object. I removed this part anyway since it's not needed anymore.